backend/infrahub/core/query/node.py

```diff
@@ -929,15 +948,15 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
             await self._add_profile_rollups(field_attribute_requirements=field_attribute_requirements)
 
         self._add_final_filter(field_attribute_requirements=field_attribute_requirements)
-        self.order_by = []
-        for far in field_attribute_requirements:
-            if not far.is_order:
-                continue
-            if far.supports_profile:
-                self.order_by.append(far.final_value_query_variable)
-                continue
-            self.order_by.append(far.node_value_query_variable)
-        self.order_by.append("n.uuid")
+        if should_order:
```

Comment:
even if the order is disabled we should also order by uuid to guarantee the pagination

Reply:
Ok I did not have that in mind

Reply:
Some additional context for the `self.order_by.append("n.uuid")`: https://github.com/opsmill/infrahub/pull/4704